x/emissions/keeper/keeper.go

```diff
@@ -1661,6 +1660,9 @@ func (k *Keeper) GetDelegateRewardPerShare(ctx context.Context, topicId TopicId,
 // Set the share on specific reputer and topicId
 func (k *Keeper) SetDelegateRewardPerShare(ctx context.Context, topicId TopicId, reputer ActorId, share alloraMath.Dec) error {
 	key := collections.Join(topicId, reputer)
+	if share.IsNaN() || !share.IsFinite() {
```

Comment:
in the types package there is a function `validateDec` - you may need to uppercase it to export it but I would advise using that function rather than this in case we want to add additional requirements in the future

Reply:
makes sense. 
And there is an additional benefit of exporting validations: that they can be used in allora-offchain-node too.